stdlib/LinearAlgebra/src/bidiag.jl

```diff
@@ -739,85 +728,205 @@ end
+    A = ldiv!(zeros(T, size(U)), B, U)
+    return B.uplo == 'U' ? UpperTriangular(A) : A
 end
-\(tA::Transpose{<:Any,<:Bidiagonal}, B::AbstractVecOrMat) = ldiv!(tA, copy(B))
-function \(adjA::Adjoint{<:Number,<:Bidiagonal{<:Number}}, B::AbstractVecOrMat{<:Number})
-    A = adjA.parent
-    TA, TB = eltype(A), eltype(B)
-    TAB = typeof((zero(TA)*zero(TB) + zero(TA)*zero(TB))/one(TA))
-    ldiv!(adjoint(convert(AbstractArray{TAB}, A)), copy_oftype(B, TAB))
+function \(B::Bidiagonal{<:Number}, L::LowerOrUnitLowerTriangular{<:Number})
+    T = typeof((oneunit(eltype(B)))\oneunit(eltype(L)))
+    A = ldiv!(zeros(T, size(L)), B, L)
+    return B.uplo == 'L' ? LowerTriangular(A) : A
+end
+
+# if we don't want the type split, the following two/three methods can be removed
+function \(U::UpperOrUnitUpperTriangular{<:Number}, B::Bidiagonal{<:Number})
+    T = typeof((oneunit(eltype(U)))/oneunit(eltype(B)))
+    A = ldiv!(U, copy_similar(B, T))
+    return B.uplo == 'U' ? UpperTriangular(A) : A
+end
+function \(L::LowerOrUnitLowerTriangular{<:Number}, B::Bidiagonal{<:Number})
+    T = typeof((oneunit(eltype(L)))/oneunit(eltype(B)))
+    A = ldiv!(L, copy_similar(B, T))
+    return B.uplo == 'L' ? LowerTriangular(A) : A
+end
+### Diagonal specialization
+function \(B::Bidiagonal{<:Number}, D::Diagonal{<:Number})
+    T = typeof((oneunit(eltype(B)))\oneunit(eltype(D)))
+    A = ldiv!(zeros(T, size(D)), B, D)
+    return B.uplo == 'U' ? UpperTriangular(A) : LowerTriangular(A)
 end
-\(adjA::Adjoint{<:Any,<:Bidiagonal}, B::AbstractVecOrMat) = ldiv!(adjA, copy(B))
+
+function _rdiv!(C::AbstractMatrix, A::AbstractMatrix, B::Bidiagonal)
+    require_one_based_indexing(C, A, B)
+    m, n = size(A)
+    if size(B, 1) != n
+        throw(DimensionMismatch("right hand side B needs first dimension of size $n, has size $(size(B,1))"))
+    end
+    if B.uplo == 'L'
+        diagB = B.dv[n]
+        for i in 1:m
+            C[i,n] = A[i,n] / diagB
+        end
+        for j in n-1:-1:1
+            diagB = B.dv[j]
+            offdiagB = B.ev[j]
+            for i = 1:m
+                C[i,j] = (A[i,j] - C[i,j+1]*offdiagB)/diagB
+            end
+        end
+    else
+        diagB = B.dv[1]
+        for i in 1:m
+            C[i,1] = A[i,1] / diagB
+        end
+        for j = 2:n
+            diagB = B.dv[j]
+            offdiagB = B.ev[j-1]
+            for i = 1:m
+                C[i,j] = (A[i,j] - C[i,j-1]*offdiagB)/diagB
+            end
+        end
+    end
+    C
+end
+rdiv!(A::AbstractMatrix, B::Bidiagonal) = @inline _rdiv!(B, A, B)
+rdiv!(A::AbstractMatrix, B::Adjoint{<:Any,<:Bidiagonal}) = rdiv!(A, copy(B))
+rdiv!(A::AbstractMatrix, B::Transpose{<:Any,<:Bidiagonal}) = rdiv!(A, copy(B))
+
+function /(A::AbstractMatrix{<:Number}, B::Bidiagonal{<:Number})
+    TA, TB = eltype(A), eltype(B)
+    TAB = typeof((oneunit(TA))/oneunit(TB))
+    _rdiv!(zeros(TAB, size(A)), A, B)
+end
+/(A::AbstractMatrix, B::Bidiagonal) = _rdiv!(copy(A), A, B)
```

Comment:
This method is for non-Number eltypes. For Number eltypes we have type promotion (right above it), otherwise (arrays of arrays) we hit the `zero(::Matrix)` wall.